### PR TITLE
chore: non-obvious type fixes not concerning IMA

### PR DESCRIPTION
**Pull request by mdrocco**
This PR provides a batch of type fixes that are:
- non-trivial from the perspective of code behavior
- not concerning IMA (I will create a separate PR for those)

In my local testing setup, they run clean all tests including regression.

### CODE REVIEW

**Review by aplanas — APPROVED** [review]
Just one nit

File: keylime/tpm/tpm_main.py [hunk 1]
```diff
@@ -303,6 +305,9 @@ def __create_ek(self, asym_alg=None):
                     "-P",
                     owner_pw,
                 ]
+            else:
+                logger.error("Internal error")
+                sys.exit()
```
Comment (aplanas) [inline keylime/tpm/tpm_main.py:310]: Conflict with #1176

Comment (mdrocco) [reply keylime/tpm/tpm_main.py:310]: This is a leftover, I will fix it. Thanks!

File: keylime/cloud_verifier_common.py [hunk 2]
```diff
@@ -321,8 +321,14 @@ def validate_ima_policy_data(agent_data):
...
+    is_valid, _, err_msg_from_validator = validators.valid_exclude_list(lists.get("exclude"))
     if not is_valid:
-        err_msg += " Exclude list regex is misformatted. Please correct the issue and try again."
+        if err_msg_from_validator is not None:
+            err_msg_base = err_msg_from_validator + " "
+        else:
+            err_msg_base = ""
+        err_msg = err_msg_base + "Exclude list regex is misformatted. Please correct the issue and try again."
+    else:
+        err_msg = err_msg_from_validator
```
[the remark this answers is not in the document]
Comment (mdrocco) [reply keylime/cloud_verifier_common.py:324]: What you mean by "usual approach"? I tried modifying as little as possible the structure of the code, feel free to propose an alternative style. Thanks!

**Comment by THS-on** [comment]
@mdrocco once the commits are squashed into a single one this LGTM.

**Review by THS-on — APPROVED** [review]

**Pull request merged**